app/domain/authentication/constraints/required_constraint.rb

```diff
@@ -0,0 +1,23 @@
+require 'command_class'
+
+module Authentication
+  module Constraints
+
+    RequiredConstraint = CommandClass.new(
```

Comment:
please call command classes with the action it performs. the name `RequiredConstraint` indicates a value object